### PR TITLE
[coordinator] add mapping rule m3 tag to drop timestamp

### PR DESCRIPTION
**Pull request by shreyassrivatsan**
**What this PR does / why we need it**:
This change adds a new m3 special tag `__m3_drop_timestamp` supported by mapping rules that indicate we should drop the timestamp and send metrics to be aggregated untimed..

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:
None

**Does this PR require updating code package or user-facing documentation?**:
None

### CODE REVIEW

File: src/cmd/services/m3coordinator/downsample/downsampler.go [hunk 1]
```diff
@@ -80,11 +80,13 @@ type SamplesAppenderOverrideRules struct {
 // SamplesAppender is a downsampling samples appender,
 // that can only be called by a single caller at a time.
 type SamplesAppender interface {
-	AppendCounterSample(value int64) error
-	AppendGaugeSample(value float64) error
-	AppendCounterTimedSample(t time.Time, value int64) error
-	AppendGaugeTimedSample(t time.Time, value float64) error
-	AppendTimerTimedSample(t time.Time, value float64) error
+	AppendUntimedCounterSample(value int64) error
```
Comment (shreyassrivatsan) [inline src/cmd/services/m3coordinator/downsample/downsampler.go:83]: These are only used by tests. I am going to check if they are even needed and remove them - maybe in a subsequent change. Will add a note reflecting the same.

File: src/cmd/services/m3coordinator/downsample/downsampler.go [hunk 2]
```diff
@@ -80,11 +80,13 @@ type SamplesAppenderOverrideRules struct {
...
+	AppendUntimedGaugeSample(value float64) error
+	AppendUntimedTimerSample(value float64) error
+	AppendCounterSample(t time.Time, value int64) error
+	AppendGaugeSample(t time.Time, value float64) error
+	AppendTimerSample(t time.Time, value float64) error
+	DropTimestamp() bool
```
Comment (wesleyk) [inline src/cmd/services/m3coordinator/downsample/downsampler.go:89]: is this only used by tests as well? The multi-appender impl is a bit confusing (in that it returns true, but not all underlying appenders will actually drop the timestamp?

**Review by wesleyk — APPROVED** [review]
LGTM!

File: src/cmd/services/m3coordinator/downsample/downsampler.go (hunk 2 above)
Comment (shreyassrivatsan) [reply src/cmd/services/m3coordinator/downsample/downsampler.go:89]: Yeah. I could not think of another way to write a test that validates which path metrics end up, other than on the appenders themselves. Let me think a bit more if I have a way to get a targeted test and then I can clean that up.

**Pull request merged**